### PR TITLE
Improve performance of gazelle --mode=diff

### PR DESCRIPTION
**Pull request by yasushi-saito**
Don't invoke the "diff" command if the contents haven't changed.
Don't call Sync() on every write. Just closing the file suffices.

Before this change, diffing a repo with 1600 BUILD files would take 60 seconds.
After the change, it takes 2 seconds.

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by yasushi-saito** [comment]
Signed the CLA.

File: go/tools/gazelle/gazelle/diff.go [hunk 1]
```diff
@@ -25,19 +26,31 @@ import (
 )
 
 func diffFile(c *config.Config, file *bf.File) error {
+	var oldContents []byte
```
Comment (jayconrod) [inline go/tools/gazelle/gazelle/diff.go]: This block can replaced with a call to `ioutil.ReadFile`.

File: go/tools/gazelle/gazelle/diff.go [hunk 2]
```diff
@@ -25,19 +26,31 @@ import (
...
+		return nil
+	}
 	f, err := ioutil.TempFile("", c.DefaultBuildFileName())
 	if err != nil {
 		return err
 	}
 	defer os.Remove(f.Name())
-	defer f.Close()
-	if _, err := f.Write(bf.Format(file)); err != nil {
+	if _, err := f.Write(newContents); err != nil {
```
Comment (jayconrod) [inline go/tools/gazelle/gazelle/diff.go]: And this can be replaced with `ioutil.WriteFile`.

**Pull request closed without merging**